app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,19 @@
+class AuthenticationController < ApplicationController
+  protect_from_forgery unless: -> { request.format.json? }
+
+  ALLOWED_DATA = %(email password).freeze
+
+  # POST /auth/login
+  def login
+    data = json_payload.select { |allow| ALLOWED_DATA.include?(allow) }
+    @user = User.find_by_email(data[:email])
+    if @user.valid_password?(data[:password])
+      token = JsonWebToken.encode(user_id: @user.id)
+      time = Time.now + 24.hours.to_i
```

Comment:
I had an **(uninitialized constant JsonWebToken::JWT)**  error that pointed to line 11, when I tried to login.

Reply:
Sorry for that. I forgot to add tests for unregistered emails.